lib/server.dart

```diff
@@ -0,0 +1,112 @@
+import 'dart:convert';
+import 'package:meta/meta.dart';
+import 'package:xml/xml.dart';
+import 'package:xml_rpc/src/converter_extension.dart';
+
+import 'src/common.dart';
+import 'src/converter.dart';
+
+/// A [MethodFailureSignature] contains the name, parameters, and error from any exception or error during a function call
+///
+/// Name of the method being called
+/// Parameters that were passed in
+/// Exception or error that occurred
+typedef MethodFailureSignature = Fault Function(String, List<dynamic>, dynamic);
+
+/// A [XmlRpcHandler] handles the handling RPC functions along with marshalling the arguments and results to / from XMLRPC spec
+///
+/// Has a set of [codecs] for encoding and decoding the datatypes
+/// To register a method or function with the dispatcher call [registerFunction] along with the function and name
+class XmlRpcHandler {
+  /// The [codecs] used for encoding and decoding
+  List<Codec> codecs;
+
+  /// The function registry
+  final Map<String, Function> methods;
+
+  /// A function that gets called on a method failure with the following data
+  ///
+  /// Name of the method being called
+  /// Parameters that were passed in
+  /// Exception or error that occurred
+  MethodFailureSignature methodFailureHandler;
+
+  /// The error code for a method failure
+  final int methodFailureCode;
+
+  /// The error code for a missing method
+  final int noExistingMethodCode;
+
+  /// Creates a [XmlRpcHandler] with the set of [codecs] for encoding and decoding
+  XmlRpcHandler({
+    @required this.methods,
+    List<Codec> codecs,
+    MethodFailureSignature methodFailureHandler,
+    this.methodFailureCode,
+    this.noExistingMethodCode,
+  }) : codecs = codecs ?? standardCodecs {
+    this.methodFailureHandler =
+        methodFailureHandler ?? defaultMethodFailureHandler;
+  }
+
+  Fault defaultMethodFailureHandler(
```

Comment:
we should make it static private by using a hard-coded code (like `-32500` - from http://xmlrpc-epi.sourceforge.net/specs/rfc.fault_codes.php linked in https://stackoverflow.com/a/2045130/634863)